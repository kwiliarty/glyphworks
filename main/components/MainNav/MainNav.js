import React from 'react'
import styled from 'styled-components'

import { NavLink } from 'react-router-dom'

const StyledNav = styled.nav`
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  justify-content: center;
  background-color: ${ props => props.theme.colors.lightBrown };
  width: 100%;
`

const StyledNavElement = styled.div`
  font-size: 1.4rem;
  margin: 0 0.5rem;
  a, a:hover, a:focus, a:active, a:visited {
    color: ${ props => props.theme.colors.darkRed };
    display: block;
    margin-top: 3px;
    padding-bottom: 4px;
    position: relative;
    text-decoration: none;
    &:focus {
      outline: 2px dotted ${ props => props.theme.colors.darkRed };
    }
    &::after {
      display: block;
      height: 0;
      width: 100%;
      background-color: ${ props => props.theme.colors.darkRed };
      content: '';
      position: absolute;
      bottom: 0;
      transition: height 0.1s linear;
    }
    &.active::after {
      height: 4px;
    }
    &:hover::after {
      height: 6px;
    }
  }
`

const MainNav = () => {
  return(
    <StyledNav>
      <StyledNavElement>
        <NavLink to='/'>Home</NavLink>
      </StyledNavElement>
      <StyledNavElement>
        <NavLink to='/glyphs'>Glyphs</NavLink>
      </StyledNavElement>
    </StyledNav>
  )
}

export default MainNav
