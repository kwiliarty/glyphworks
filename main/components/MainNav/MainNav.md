The main application navigation.

```jsx
<MainNav />
```
