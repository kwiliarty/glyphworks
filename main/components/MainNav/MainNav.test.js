import React from 'react'
import { render, screen } from 'utils'
import MainNav from './MainNav'

describe( 'The MainNav component', () => {

  it( 'displays links to top-level pages', () => {
    render( <MainNav /> )
    const nav = screen.getByRole( 'navigation' )
    expect( nav ).toHaveTextContent( 'Home' )
    expect( nav ).toHaveTextContent( 'Glyphs' )
    expect( nav ).toMatchSnapshot()
  })
})
