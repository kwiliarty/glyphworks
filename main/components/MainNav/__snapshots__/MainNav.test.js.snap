// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The MainNav component displays links to top-level pages 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  background-color: #c9b978;
  width: 100%;
}

.c1 {
  font-size: 1.4rem;
  margin: 0 0.5rem;
}

.c1 a,
.c1 a:hover,
.c1 a:focus,
.c1 a:active,
.c1 a:visited {
  color: #621818;
  display: block;
  margin-top: 3px;
  padding-bottom: 4px;
  position: relative;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c1 a:focus,
.c1 a:hover:focus,
.c1 a:focus:focus,
.c1 a:active:focus,
.c1 a:visited:focus {
  outline: 2px dotted #621818;
}

.c1 a::after,
.c1 a:hover::after,
.c1 a:focus::after,
.c1 a:active::after,
.c1 a:visited::after {
  display: block;
  height: 0;
  width: 100%;
  background-color: #621818;
  content: '';
  position: absolute;
  bottom: 0;
  -webkit-transition: height 0.1s linear;
  transition: height 0.1s linear;
}

.c1 a.active::after,
.c1 a:hover.active::after,
.c1 a:focus.active::after,
.c1 a:active.active::after,
.c1 a:visited.active::after {
  height: 4px;
}

.c1 a:hover::after,
.c1 a:hover:hover::after,
.c1 a:focus:hover::after,
.c1 a:active:hover::after,
.c1 a:visited:hover::after {
  height: 6px;
}

<nav
  class="c0"
>
  <div
    class="c1"
  >
    <a
      aria-current="page"
      class="active"
      href="/"
    >
      Home
    </a>
  </div>
  <div
    class="c1"
  >
    <a
      class=""
      href="/glyphs"
    >
      Glyphs
    </a>
  </div>
</nav>
`;
